src/off_chain/historical_data/daemon_analytics.py

```diff
@@ -0,0 +1,40 @@
+import time
+from typing import List
+from cow_endpoint_surplus import EBBOHistoricalDataTesting
+from web3 import Web3
+import directory
+import logging
+
+
+def main() -> None:
+    Instance = EBBOHistoricalDataTesting()
+    infura_connection = f"https://mainnet.infura.io/v3/{directory.INFURA_KEY}"
+    w3 = Web3(Web3.HTTPProvider(infura_connection))
+    start_block = w3.eth.block_number
+    time.sleep(1)
```

Comment:
What is the purpose of this sleep? are you trying to make `start_block != end_block`? 

I see another sleep at the top of the while loop that probably belongs in line 35 (immediately after the "going to sleep" log), but I am also not so sure this approach is ideal... there is a new EVM block published every 12 -15 seconds... this means that the program will wake up too often and start block will often equal end block. causing this entire block of code to run (and make requests) at least 12 times more than necessary. 

If sleep statement the way this is going to be approached I would suggest 60 seconds (~5 blocks).

Reply:
I noticed that I pushed the sleep for one second (which is a debug test), but the final version is going to be asleep for 30 mins (1800 secs). I've added a comment at the top of main in the push (which I'm going to be making soon) that describes the exact behavior, but let me just put it here as well.

"""
_At runtime, this function records the most recent block and initializes it as the start block.
After the daemon is asleep for 30 mins, it gets the newest block as the end block. 
Since the competition endpoint has a lag of 30 mins (worst case), we wait 30 mins before we fetch comp. data
and start checking for potential surplus. 
If the data was not retrievable for any reason, it adds it to the list of unchecked hashes 
to be checked in the next cycle. Once all hashes in the current cycle have been iterated through, 
the previous end block + 1 becomes the start block for the next cycle, and the latest block is the new end block.
Daemon sleeps for 30 mins and continues checking._
"""

Reply:
If its only every 30 minutes then it might make more sense for this to be a cronjob. That is, cheaper than running an idling service 24/7

Reply:
I actually thought about the sleep time, in relation to some other discussion we also had with @bh2smith and it seems to me that we should actually reduce it after all to something like 5 minutes. This will ensure that we will most likely not get rate-limited from the competition endpoint, as the queries will be a few every few minutes instead of coming into bursts. It will also allow us to have better response times and get warnings faster, if needed.

Reply:
Another suggestion here is to make the sleep time into a runtime configuration parameter.